meta-mbl-distro/scripts/lib/update/payload.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import logging
+import pathlib
+import subprocess
+import tempfile
+
+from update.util import get_bitbake_conf_var
+import update.images as images
+import update.swdesc as swdesc
+import update.testinfo as testinfo
+
+CURRENT_PAYLOAD_FORMAT_VERSION = 3
+
+PAYLOAD_FORMAT_VERSION_INFO = {
```

Comment:
I'm not sure I like this apparent "constant" returning a dictionary. Could we make this a function call  instead? Python doesn't have  "const" or anything, but the convention is that these uppercase variables are "constant" values so it feels a bit misleading the make this a dictionary.

Reply:
But it's a constant dictionary. I'm trying to convey that the dictionary won't change after it has been constructed.

Reply:
I'm not sure why this nested dictionary is required anyway. What does having this "image_format_versions" key buy us rather than having a tuple of "image format versions"? Are we planning on extending the inner dictionary with more keys? I kind of see the need for the outer dictionary if we're planning on handling multiple payload format versions, but I don't get why we need the inner dict.

Reply:
I think an example will illustrate what I'm thinking best. Suppose we introduce a new format for rootfs images (e.g. an ext4 image file rather than a tar). Then I'd expect to update `PAYLOAD_FORMAT_VERSION _INFO` to look like this:
```Python
PAYLOAD_FORMAT_VERSION_INFO = {
    3: {
        "image_format_versions": {
            images.MBL_WKS_BOOTLOADER1_ID: 3,
            images.MBL_WKS_BOOTLOADER2_ID: 3,
            images.MBL_BOOT_ID: 3,
            images.MBL_ROOTFS_ID: 3,
            images.MBL_APPS_ID: 3,
        }
    },
    4: {
        "image_format_versions": {
            images.MBL_WKS_BOOTLOADER1_ID: 3,
            images.MBL_WKS_BOOTLOADER2_ID: 3,
            images.MBL_BOOT_ID: 3,
            images.MBL_ROOTFS_ID: 4,
            images.MBL_APPS_ID: 3,
        }
    }
}
```
 Payload format version 4 would have the same image format versions as payload version 3, except for the rootfs image format version which would be 4.

I think I made a mistake in adding this dict  before we actually have a mechanism for choosing different payload format versions though. It doesn't really make much sense on its own. I'll remove it for now.

Reply:
Oh, I think I misunderstood your question... Either way, I think it will be addressed by me removing the dict.

Reply:
Addressed in https://github.com/ARMmbed/meta-mbl/pull/763/commits/174d5ce546fed12c21052193ef8922b975c4d28e.